packages/shopping/src/datasources/mongo.datasource.ts

```diff
@@ -3,17 +3,34 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {inject} from '@loopback/core';
+import {inject, lifeCycleObserver, ValueOrPromise} from '@loopback/core';
 import {juggler, AnyObject} from '@loopback/repository';
 const config = require('./mongo.datasource.json');
 
+function updateConfig(dsConfig: AnyObject) {
+  if (process.env.KUBERNETES_SERVICE_HOST) {
+    dsConfig.host = process.env.SHOPPING_APP_MONGODB_SERVICE_HOST;
+    dsConfig.port = +process.env.SHOPPING_APP_MONGODB_SERVICE_PORT!;
+  }
+  return dsConfig;
+}
+
+@lifeCycleObserver('datasource')
```

Comment:
Very cool use of life cycle observers.

Reply:
We now generate datasources in this pattern with `lb4 datasource`.